### PR TITLE
Clean lexer

### PR DESCRIPTION
**Pull request by forki**
* Removed unused permitFsharpKeywords switch
* cleaned up pattern matching

### CODE REVIEW

File: src/fsharp/lexhelp.fs [hunk 1]
```diff
@@ -305,48 +305,44 @@ module Keywords =
...
-            IdentifierToken args lexbuf s
-        else
-          let mutable v = Unchecked.defaultof<_>
-          if keywordTable.TryGetValue(s, &v) then 
-              if (match v with RESERVED -> true | _ -> false) then
-                  warning(ReservedKeyword(FSComp.SR.lexhlpIdentifierReserved(s), lexbuf.LexemeRange));
-                  IdentifierToken args lexbuf s
-              else v
-          else 
+        match keywordTable.TryGetValue s with
```
Comment (dsyme) [inline src/fsharp/lexhelp.fs]: Please double check this return-a-tuple-and-match-on-it formulation produces code at least as good as before.  I still tend to write the mutable by hand in performance-critical code but I know there is an optimization that should give us the right code here.

**Comment by forki** [comment]
Somehow the build is broken (unrelated to this)

    Installed:
        93 package(s) to C:\projects\visualfsharp-3dtit\lkg\project.json
    Error: dotnet restore failed

#sigh

File: src/fsharp/lexhelp.fs (hunk 1 above)
Comment (forki) [inline src/fsharp/lexhelp.fs]: internal static Parser.token KeywordOrIdentifierToken(Lexhelp.lexargs args, LexBuffer<char> lexbuf, string s)
      {
        Parser.token token1 = (Parser.token) null;
        bool flag = Lexhelp.Keywords.keywordTable.TryGetValue(s, out token1);
        Parser.token token2 = token1;
        if (flag)
        {
             ...
        }
      }

**Comment by dsyme** [comment]
Thanks for checking!

**Comment by dsyme** [comment]
@KevinRansom We can merge this once CI is fixed

**Comment by KevinRansom** [comment]
@forki Thank you for taking care of this

**Pull request merged**